src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js

```diff
@@ -50,6 +44,26 @@ class AvailableOrganisationUnitsTree extends PureComponent {
         }
     }
 
+    async loadAvailableOrgUnits() {
+        const d2 = this.context.d2
+
+        const orgUnits = await d2.currentUser.getDataViewOrganisationUnits({
+            fields: 'id,displayName,path,children::isNotEmpty,memberCount',
+            paging: false,
+        })
+
+        if (!orgUnits.size && d2.currentUser.authorities.has('ALL')) {
+            // if special all-authority we have access to all orgunits
+            return d2.models.organisationUnits.list({
+                paging: false,
+                level: 1,
+                fields: 'id,displayName,path,children::isNotEmpty,memberCount',
+            })
+        }
```

Comment:
I'm not 100% sure this is always going to behave correctly. If the goal is to present users with the `ALL` authority with all the org-units, which I think is correct, then I would consider changing this a little bit... Maybe something like this:
```javascript
const fields = 'id,displayName,path,children::isNotEmpty,memberCount';

return d2.currentUser.authorities.has('ALL')
    ? d2.models.organisationUnits.list({ paging: false, level: 1, fields })
    : d2.currentUser.getDataViewOrganisationUnits({ fields, paging: false, })

```

Reply:
Why would it not behave correctly? It is only for admin users that does not have assigned orgunits. Eg. `system`-user. If the user has assigned org-units and has the all-authority we should just use the assigned org-units, at least that's how it has worked in the maintenance app for ever.

Reply:
> If the user has assigned org-units and has the all-authority we should still use the assigned org-units, at least that's how it has worked in the maintenance app forever.

OK, if this is the case, the current code makes perfect sense.

Reply:
I will update the comment with a note for that.